proxies/live/apiproxy/targets/ers-target.xml

```diff
@@ -15,29 +15,38 @@
             </Step>
             <Condition>(fault.name Matches "SpikeArrestViolation")</Condition>
         </FaultRule>
+        <FaultRule name="quota_fault">
+            <Step>
+                <Name>AssignMessage.QuotaErrorResponse</Name>
+            </Step>
+            <Condition>(fault.name Matches "QuotaViolation")</Condition>
+        </FaultRule>
     </FaultRules>
     <PreFlow>
         <Request>
             <Step>
                 <Name>OauthV2.VerifyAccessToken</Name>
             </Step>
             <Step>
-                <!-- Should use Java Regex for non binary request -->
-                <Condition>(proxy.pathsuffix Not Matches "/**/Binary/*")</Condition>
-                <Name>SpikeArrest</Name>
+                <Name>AssignMessage.PopulateAsidFromApp</Name>
             </Step>
             <Step>
-                <Condition>(proxy.pathsuffix MatchesPath "/**/Binary/*")</Condition>
-                <Name>SpikeArrest.Binary</Name>
+                <Name>AssignMessage.SetAsidHeader</Name>
             </Step>
             <Step>
-                <Name>AssignMessage.PopulateAsidFromApp</Name>
+                <Name>AssignMessage.AddBaseUrlHeader</Name>
             </Step>
             <Step>
-                <Name>AssignMessage.SetAsidHeader</Name>
+                <Condition>(proxy.pathsuffix MatchesPath "/**/Binary/*")</Condition>
```

Comment:
I wonder if it's worth creating a separate flow variable to capture this condition that the other steps could depend on rather than the condition having to be duplicated here? For example:
```
<Step>
    <Name>AssignMessage.SetIsBinaryCall</Name><!-- Sets the flow variable isBinaryCall to true -->
    <Condition>proxy.pathsuffix MatchesPath "/**/Binary/*" </Condition>
</Step>
```
Other steps could then simply use `isBinaryCall` or `not(isBinaryCall)` as their conditions.

Reply:
I may have to use a Javascript policy to set this variable. I don't think I can use AssignMessage to set a flow variable depending on whether the path contains 'Binary'.